cmd/query/app/server_test.go

```diff
@@ -53,10 +62,314 @@ func TestCreateTLSServerError(t *testing.T) {
 	}
 
 	_, err := NewServer(zap.NewNop(), &querysvc.QueryService{},
-		&QueryOptions{TLS: tlsCfg}, opentracing.NoopTracer{})
+		&QueryOptions{TLSGRPC: tlsCfg}, opentracing.NoopTracer{})
 	assert.NotNil(t, err)
 }
 
+func TestCreateTLSHttpServerError(t *testing.T) {
+	tlsCfg := tlscfg.Options{
+		Enabled:      true,
+		CertPath:     "invalid/path",
+		KeyPath:      "invalid/path",
+		ClientCAPath: "invalid/path",
+	}
+
+	_, err := NewServer(zap.NewNop(), &querysvc.QueryService{},
+		&QueryOptions{TLSHTTP: tlsCfg}, opentracing.NoopTracer{})
+	assert.NotNil(t, err)
+}
+
+func TestTLSHTTPServer(t *testing.T) {
+	tests := []struct {
+		name             string
+		serverTLS        tlscfg.Options
+		clientTLS        tlscfg.Options
+		expectError      bool
+		expectServerFail bool
+	}{
+		// {
```

Comment:
remember to remove if not needed

Reply:
I left that as a template for adding future tests if necessary.  Some fields are relevant for client and not for server and vice-versa.  Should i go ahead and remove it?  thanks.

Reply:
the struct definition is sufficient, no need for template